src/lib/utils.js

```diff
@@ -15,20 +15,36 @@ export const applyImgixParameters = (source, layout, imgixProps = {}) => {
     return source
   }
 
-  const path = uri.split(/[?#]/)[0]
+  const [path, query] = uri.split(/[?#]/)
+
+  const givenParams = query && query.split('&')
+  const queryValues = {}
+
+  if (givenParams) {
+    for (const param of givenParams) {
+      if (param) {
+        const [key, value] = param.split('=')
+        queryValues[key] = value
+      }
+    }
+  }
+
   const params = {
     w: layout.width,
     h: layout.height,
     dpr: Math.max(PixelRatio.get(), 2),
     auto: Platform.OS === 'web' ? 'format,compress' : 'compress',
     ...(typeof imgixProps === 'object' && { ...imgixProps }),
+    ...queryValues,
   }
-  const queryParams = Object.keys(params).map(key => `${key}=${params[key]}`).join('&')
+  const queryParams = Object.keys(params)
```

Comment:
it already exists in the current code but query params need to be encoded when added to a URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent